sub-packages/bionemo-esm2/src/bionemo/esm2/data/datamodule.py

```diff
@@ -18,19 +18,18 @@
 import os
 from typing import Literal
 
-import pytorch_lightning as pl
-import torch
-import torch.utils.data
+from nemo.lightning.data import WrappedDataLoader
 from nemo.lightning.pytorch.plugins import MegatronDataSampler
 from nemo.utils import logging
 from pytorch_lightning.utilities.types import EVAL_DATALOADERS, TRAIN_DATALOADERS
 
 from bionemo.esm2.data import dataset, tokenizer
 from bionemo.llm.data import collate
+from bionemo.llm.data.datamodule import MegatronDatamodule
 from bionemo.llm.utils.datamodule_utils import infer_num_samples
 
 
-class ESMDataModule(pl.LightningDataModule):
```

Comment:
Can you comment on why is this change needed?

Reply:
nvm I thought its an import from megatron

Reply:
Yeah, this is super important. If you do not do this and you restart your training job it will go back to step 0 rather than picking up from where it left off! You also need to call that function when you return the dataloader. So brittle. We need to heavily document this!

Reply:
Yeah I called this `MegatronDataModule` because of how it's important in the context of the megatron strategy specifically. This DataModule class is needed to expose the functionality and state that is critical for enabling the megatron strategy to pick up where it left off in terms of sample index during training.